docs/dev/framework/csp.md

```diff
@@ -0,0 +1,229 @@
+---
+title: Content Security Policy
+description: "Contao's Content Security Policy framework."
+---
+
+{{< version "5.3" >}}
+
+Contao **5.3** introduced support for [Content Security Policies][CSPWebsite] ([MDN Web Docs][CSPMDN]) for its front
+end. Implementing CSP directives is an important tool to secure your website against XSS attacks and unwanted external
+resources. At the same time we want to specifically allow scripts and resources generated or added by your application.
+In this article we will show you how to use Contao's CSP framework in order to adjust the generated 
+`Content-Security-Policy` response header from within your templates, controllers and services.
+
+
+## The `CspHandler`
+
+Content Security Policies are applied via Contao's [Response Context][ResponseContext] concept. When CSP is enabled in
+[the settings][CspPageSettings] of a website root a `CspHandler` instance will be added to the response context. You can 
+access the `CspHandler` in your services like this for example:
+
+```php
+// src/ExampleService.php
+namespace App;
+
+use Contao\CoreBundle\Routing\ResponseContext\Csp\CspHandler;
+use Contao\CoreBundle\Routing\ResponseContext\ResponseContextAccessor;
+
+class ExampleService
+{
+    public function __construct(private readonly ResponseContextAccessor $responseContextAccessor)
+    {
+    }
+
+    public function __invoke(): void
+    {
+        $responseContext = $this->responseContextAccessor->getResponseContext();
+
+        if ($responseContext?->has(CspHandler::class)) {
+            /** @var CspHandler $csp */
+            $cspHandler = $responseContext->get(CspHandler::class);
+
+            // Retrieve nonces, add sources, …
+        }
+    }
+}
+```
+
+The following describes the most important methods of the `CspHandler`. In many cases you will want to make these
```

Comment:
You might want to do that in the code for app stuff. Maybe we should write something along the lines of "in case you are writing an extension you will most likely want to make these adjustments directly from within templates in order to give your users maximum flexibility..."?

Reply:
Okay, but the info on "why" you would want to do that in the template and not in the controller is missing ;)

Reply:
Not sure what the answer to that is. Can you make a suggestion on how to word this differently then?

Reply:
I have shortened the wording a bit, wdyt?

Reply:
Perfect, this can be merged :)